__tests__/app.test.js

```diff
@@ -149,4 +149,104 @@ describe("GET", () => {
+            .send(newComment)
+            .expect(201)
+            .then((res) => {
+                expect(res.body.comment).toMatchObject({
+                    comment_id :19,
+                    article_id: 2,
+                    created_at : expect.any(String),
+                    author : 'icellusedkars',
+                    body: 'Not a good article'
+
+                })
+            })
+    })
+
+    test("comments- return status 404 with an an error message if the id does not exist", () => {
+        const newComment = {
+            username:'icellusedkars',
+            body: 'Not a good article'
+        }
+        return request(app)
+            .post('/api/articles/50/comments')
+            .send(newComment)
+            .expect(404)
+            .then((res) => {
+                expect(res.body.msg).toBe('article not found')
+            })
+    })
+
+    test("comments- return status 400 with an an error message if the id is not valid", () => {
+        const newComment = {
+            username:'icellusedkars',
+            body: 'Not a good article'
+        }
+        return request(app)
+            .post('/api/articles/invalid/comments')
+            .send(newComment)
+            .expect(400)
+            .then((res) => {
+                expect(res.body.msg).toBe('invalid request for article id')
+            })
+    })
+})
+
+describe("PATCH", () => {
+    test("articles- return status 200 with the updated object", () => {
+        const articleUpdates = {
+            inc_votes: 2
+        }
+        return request(app)
+            .patch('/api/articles/1')
+            .send(articleUpdates)
+            .expect(200)
+            .then((res) => {
+                expect(res.body.article).toEqual({
+                    article_id:1,
+                    title: "Living in the shadow of a great man",
+                    topic: "mitch",
+                    author: "butter_bridge",
+                    body: "I find this existence challenging",
+                    created_at: expect.any(String),
+                    votes: 102,
+                })
+            })
+    })
+
+    test("articles- return status 400 with an an error message if the id is not valid", () => {
+        const articleUpdates = {
+            inc_votes: 2
+        }
+        return request(app)
+            .patch('/api/articles/invalid')
+            .send(articleUpdates)
+            .expect(400)
+            .then((res) => {
+                expect(res.body.msg).toBe('invalid request for article id')
+            })
+    })
+
+    test("articles- return status 404 with an an error message if the id is not found", () => {
+        const articleUpdates = {
+            inc_votes: 2
+        }
+        return request(app)
+            .patch('/api/articles/100')
+            .send(articleUpdates)
+            .expect(404)
+            .then((res) => {
+                expect(res.body.msg).toBe('article not found')
+            })
+    })
+})
```

Comment:
Would like to see some tests around the actual vote object. What if we send an incorrect key, or the wrong type of data for the inc_votes?